core/commands/diff.py

```diff
@@ -738,6 +744,67 @@ def jump_position_to_file(self, diff, pt):
 
         return filename, row, col
 
+    def jump_position_to_file_for_word_diff_mode(self, diff, pt):
+        # type: (ParsedDiff, int) -> Optional[Tuple[str, int, int]]
+        head_and_hunk_offsets = head_and_hunk_for_pt(diff, pt)
+        if not head_and_hunk_offsets:
+            return None
+
+        view = self.view
+        header_region, hunk_region = head_and_hunk_offsets
+        header = extract_content(view, header_region)
+        hunk = extract_content(view, hunk_region)
+        hunk_start, _ = hunk_region
+
+        # The hunk contains itself a header line e.g. "@@ -686,8 +686,14 @@ ...",
+        # so the actual content of the hunk starts at the line after that.
+        content_start = view.full_line(hunk_start).end()
+
+        # Select all "deletion" regions in the hunk up to the cursor (pt)
+        removed_regions_before_pt = [
+            # In case the cursor is *in* a region, shorten it up to
+            # the cursor.
+            sublime.Region(region.begin(), min(region.end(), pt))
+            for region in view.get_regions('git-savvy-removed-bold')
+            if content_start <= region.begin() < pt
+        ]
+
+        # Select all completely removed lines, but exclude lines
+        # if the cursor is exactly at the end-of-line char.
+        removed_lines_before_pt = sum(
+            region == view.line(region.begin()) and region.end() != pt
+            for region in removed_regions_before_pt
+        )
+        line_start = view.line(pt).begin()
+        removed_chars_before_pt = sum(
+            region.size()
+            for region in removed_regions_before_pt
+            if line_start <= region.begin() < pt
+        )
+
+        # Compute the *relative* row in that hunk
+        head_row, _ = view.rowcol(hunk_start)
+        pt_row, col = view.rowcol(pt)
+        rel_row = pt_row - (head_row + 1)
```

Comment:
Here is small bug, namely when you're on the `@@ 456,23 ...` line it will not jump to the first content line.  So it basically is accurate if the cursor is somewhere on the content. 

Probably

```suggestion
        head_row, _ = view.rowcol(content_start)
        pt_row, col = view.rowcol(pt)
        rel_row = max(0, pt_row - head_row)
```

Reply:
Done